autosubmitconfigparser/config/configcommon.py

```diff
@@ -1634,6 +1654,7 @@ def reload(self, force_load=False, only_experiment_data=False, save=False):
             self.current_loaded_files = {}  # reset loaded files
             for filename in self.get_yaml_filenames_to_load(self.conf_folder_yaml):
                 starter_conf = self.unify_conf(starter_conf, self.load_config_file(starter_conf, Path(filename)))
+            starter_conf = self.load_as_env_variables(starter_conf)
             starter_conf = self.load_common_parameters(starter_conf)
             self.starter_conf = starter_conf
```

Comment:
For later, but I think these functions don't need to return a dict... if they are only modifying the contents of the dict, we can pass it and they will be modified as the functions get references?

Reply:
Yes, however, I'm not sure if there are scenarios where it would be beneficial to return the dictionary.

I have considered it many times in the past, but I didn't do it mainly because implementing this change would require revising how it is called in the testing suite and Autosubmit, so it should be addressed as a separate issue.